### ISSUE TITLE
Setting a class property with a different indentation, changes indentation rule

### ISSUE THREAD

**Issue by larsmunkholm** [eslint/eslint#11485]
**My environment**

* **ESLint Version:** 5.14.1
* **Node Version:** 10.15.1
* **npm Version:** 6.4.1

**Parser** babel-eslint with plugin `@babel/plugin-proposal-class-properties`

**My full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true
  },
  "plugins": [
    "react"
  ],
  "parserOptions": {
    "sourceType": "module",
    "allowImportExportEverywhere": true,
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    }
  },
  "rules": {
    "indent": ["warn", 4, {"SwitchCase": 1}],
    "linebreak-style": ["off", "unix"],
    "quotes": ["off", "double"],
    "semi": ["error", "always" ],
    "prefer-const" : ["error"],
    "no-new-object" : ["error"],
    "object-shorthand" : ["error"],
    "quote-props" : ["error", "consistent"],
    "prefer-template" : ["error"],
    "template-curly-spacing" : ["error"],
    "no-useless-escape" : ["error"],
    "prefer-rest-params" : ["error"],
    "no-param-reassign" : ["error"],
    "prefer-spread" : ["error"],
    "prefer-arrow-callback" : ["error"],
    "arrow-spacing" : ["error"],
    "arrow-parens" : ["warn", "as-needed"],
    "arrow-body-style" : ["error"],
    "no-duplicate-imports" : ["error"],
    "dot-notation" : ["error"],
    "no-plusplus" : ["error"],
    "no-unneeded-ternary" : ["error"],
    "space-before-blocks" : ["warn"],
    "keyword-spacing" : ["warn"],
    "space-infix-ops" : ["warn"],
    "eol-last" : ["warn"],
    "newline-per-chained-call" : ["warn"],
    "space-in-parens" : ["error"],
    "array-bracket-spacing" : ["error"],
    "camelcase" : ["error"],
    "new-cap" : ["error"],
    "eqeqeq" : ["error"],
    "no-var" : ["error"]
  },
  "settings": {
    "react": {
      "createClass": "createReactClass",
      "pragma": "React",
      "version": "16.4",
      "flowVersion": "0.79"
    },
    "propWrapperFunctions": [ "forbidExtraProps" ]
  }
}
```
</details>

 ...

**The actual source code causing the issue**

```js
export default class CartSummary extends PureComponent {
    constructor() {
        super();
        this.state = {
            showPromotions: false
        };
    }

  propTypes = { // <- Setting a class property with a different indentation, changes indentation rule
      itemCount: PropTypes.number.isRequired,
      subTotalFormatted: PropTypes.string.isRequired,
      isGuestUser: PropTypes.bool.isRequired,
      hasGuestCheckout: PropTypes.bool.isRequired,
      onSignInClicked: PropTypes.func.isRequired,
      totalPriceFormatted: PropTypes.string.isRequired,
      shippingTotal: PropTypes.string.isRequired,
      showDeliveryPrice: PropTypes.bool.isRequired,
      amountSavedFormatted: PropTypes.string,
      hasGeneralCartError: PropTypes.bool,
      hasAnyCartError: PropTypes.bool,
      errorMessage: PropTypes.string
  };

    static defaultProps = {
        hasGeneralCartError: false,
        hasAnyCartError: false,
        errorMessage: "",
        amountSavedFormatted: ""
    };

    handleShowPromotions() {
        this.setState(prevState => ({
            showPromotions: !prevState.showPromotions
        }));
    }

    render() { ... }
}
```

```bash
eslint **/*.js
```

**Expectation**
I expected to get some warnings about wrong indentation in the `propTypes` property. It should be 4 spaces.

**What actually happened?**
No warnings regarding indentation in `propTypes` ... But lots of warnings in the code following propTypes. Eslint wants an indentation of 2 spaces now, even though the rules are set to 4 spaces. When a class property is indented with a different number of spaces, the rule changes for the rest of the code in the scope.

**Comment by larsmunkholm** [comment]
This can be fixed by adding `"ClassProperty"` to `KNOWN_NODES` in lib/rules/indent.js

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. Class properties are experimental syntax (they are a [stage 3 proposal](https://github.com/tc39/proposal-class-fields)), so we don't support them in core rules. I would recommend creating a custom rule if you'd like the `indent` rule to work well with class properties (you could probably copy the existing rule and just add `ClassProperty` to the list).
